client-v2/src/components/FrontsCAPIInterface/SearchInput.js

```diff
@@ -42,47 +89,140 @@ class FrontsCAPISearchInput extends React.Component<
     });
   };
 
-  handleTagInput = (item: any) => {
+  handleTagSearchInput = (
+    { currentTarget }: SyntheticEvent<HTMLInputElement>,
+    type: string
+  ) => {
+    const termInState = `${type}SearchTerm`;
     this.setState({
-      tag: item ? item.id : null
+      [termInState]: currentTarget.value
     });
   };
 
+  handleTagInput = (item: any, type: string) => {
+    const searchTerm = `${type}SearchTerm`;
+    const newTags = this.state[type];
```

Comment:
`newTags` is being mutated here and when it goes back in the state the array is the same as when it came out, even if new stuff has been added. Perhaps you could use `let newTags` here and `newTags = newTags.concat(...)` below to make sure that if we use `===` to check for equality with this array somewhere, that it'll work as expected.